step-capstone/src/components/Trip.js

```diff
@@ -112,80 +21,71 @@ export default class Trip extends React.Component {
         this.handleAddItem = this.handleAddItem.bind(this);
     }
 
-    fetchData() {
-        // TODO: fetch from datastore
-        return testData;
-    }
-
     componentDidMount() {
-        let data = this.fetchData();
-        this.setState({
-            items: data
-        })
+      let travelObjectList = [];
+      this.context.getTrip(this.state.reference)
+      .then(data => {
+        data.data().travelObjects.forEach(travelObject => {
+          travelObjectList.push(travelObject)
+        });
+        this.setState({items : travelObjectList});
+      })
+      .catch(error => {
+        console.log("Error Getting Trip Data")
+        console.error(error)
```

Comment:
Logging the error seems like a good approach. Even better would be surfacing an error back to the user, but I think this makes sense for the MVP.

Reply:
yes I agree. all the error handling right now is pretty weak